apollo-router/src/query_planner/bridge_query_planner_pool.rs

```diff
@@ -250,18 +288,16 @@ impl tower::Service<QueryPlannerRequest> for BridgeQueryPlannerPool {
         &mut self,
         _cx: &mut std::task::Context<'_>,
     ) -> std::task::Poll<Result<(), Self::Error>> {
-        if self.sender.is_full() {
-            std::task::Poll::Ready(Err(QueryPlannerError::PoolProcessing(
-                "query plan queue is full".into(),
-            )))
-        } else {
-            std::task::Poll::Ready(Ok(()))
+        match &self.pool_mode {
+            PoolMode::Pool { sender, .. } if sender.is_full() => std::task::Poll::Ready(Err(
+                QueryPlannerError::PoolProcessing("query plan queue is full".into()),
+            )),
+            _ => std::task::Poll::Ready(Ok(())),
```

Comment:
As discussed, we need a way to set a bound on this. I'd be happy with a hard-coded limit (semaphore controlled maybe, perhaps not configurable) until we have a better mechanism.

Reply:
If we also merge https://github.com/apollographql/router/pull/6122 then in pass-through mode we could generate backpressure if [`thread_poll.queued_count()`](https://docs.rs/threadpool/latest/threadpool/struct.ThreadPool.html#method.queued_count) is above some threshold

By the way, backpressure be generated by returning `Poll::Pending` rather than `Poll::Ready(Err(…))`?

https://docs.rs/tower/latest/tower/trait.Service.html#tymethod.poll_ready

Reply:
Yep. I think the reason it was done as a ready error is to fit in with other problems in our existing pipeline. I agree that we should fix all that stuff and make it `Poll::Pending`.

Reply:
the `queued_count()` value can also be used to generate the `apollo.router.query_planning.queued` metric